pkg/operator/ceph/cluster/controller.go

```diff
@@ -214,64 +216,182 @@ func (c *ClusterController) onK8sNodeAdd(obj interface{}) {
+		cluster.mons.ClusterInfo = cluster.Info
+	}
+
+	// Start mon health checker
+	healthChecker := mon.NewHealthChecker(cluster.mons, cluster.Spec)
+	go healthChecker.Check(cluster.stopCh)
+
+	if !cluster.Spec.External.Enable {
+		// Start the osd health checker only if running OSDs in the local ceph cluster
+		osdChecker := osd.NewMonitor(c.context, cluster.Namespace)
+		go osdChecker.Start(cluster.stopCh)
+	}
+}
+
+func (c *ClusterController) configureExternalCephCluster(namespace, name string, cluster *cluster) error {
+	// Make sure the spec contains all the information we need
+	err := validateExternalClusterSpec(cluster)
+	if err != nil {
+		return fmt.Errorf("failed to validate external cluster specs. %+v", err)
+	}
+
+	if err := c.updateClusterStatus(namespace, name, cephv1.ClusterStateConnecting, ""); err != nil {
+		logger.Warningf("failed to update cluster status in namespace %s: %+v", cluster.Namespace, err)
+	}
+
+	// loop until we find the secret necessary to connect to the external cluster
+	for {
+		var err error
+		cluster.Info, _, _, err = mon.LoadClusterInfo(c.context, namespace)
+		if err != nil {
+			logger.Warningf("waiting for the connection info of the external cluster. %+v", err)
+			time.Sleep(10 * time.Second)
+			continue
+		}
+		logger.Infof("found the cluster info to connect to the external cluster. mons=%+v", cluster.Info.Monitors)
+
+		// Write the rook-ceph-config configmap (used by various daemons to apply config overrides)
+		// If we don't do this, daemons will never start, waiting forever for this configmap to be present
+		err = config.GetStore(cluster.context, namespace, &cluster.ownerRef).CreateOrUpdate(cluster.Info)
+		if err != nil {
+			return err
+		}
+
+		// Let's write connection info (ceph config file and keyring) to the operator for health checks
+		err = mon.WriteConnectionConfig(cluster.context, cluster.Info)
+		if err != nil {
+			return fmt.Errorf("failed to write connection info %+v", err)
+		}
+
+		// Get Ceph monitors version on the external cluster
+		cephMonVersion, err := client.GetCephMonVersion(c.context, namespace)
+		if err != nil {
+			return fmt.Errorf("failed to get ceph mon version. %+v", err)
+		}
+
+		logger.Infof("detecting the image version provided for the external cluster...")
+		specCephVersionImage, err := cluster.detectCephVersion(c.rookImage, cluster.Spec.CephVersion.Image, detectCephVersionTimeout)
+		if err != nil {
+			return fmt.Errorf("unknown ceph major version. %+v", err)
+		}
+		specCephVersion := *specCephVersionImage
+
+		// Populate clusterInfo with the external cluster version
+		cluster.Info.CephVersion = *cephMonVersion
+
+		// Make sure the external cluster version and the ceph version in the provided image are identical
+		if !cephver.IsIdentical(specCephVersion, cluster.Info.CephVersion) {
+			return fmt.Errorf("wrong ceph version %s, external cluster version is %s, they must match", specCephVersion.String(), cephMonVersion.String())
```

Comment:
For upgrades I don't think we can absolutely require this. Otherwise if you update the remote cluster, this cluster's orchestration will be broken until you trigger the upgrade here as well. What if we just log an error and continue? Ideally we would expose an error status on the CR as well.

Reply:
The problem is that if we continue, we might not be able to successfully create resources on the k8s cluster.

Perhaps we can only check the major number, assuming nothing should break between 2 minor release.